src/examples/signal_generator.cpp

```diff
@@ -0,0 +1,140 @@
+#include <iostream>
+#include <hyro/common/Time.h>
+#include <hyro/core/StateMachine.h>
+#include <hyro/widgets/WidgetCollection.h>
+#include <hyro/factory/CommandFactory.h>
+#include <hyro/utils/DynamicPropertyAccess.h>
+
+#include <../include/hyro/SignalGeneratorComponent.h>
+#include <../include/hyro/DigitalConverterComponent.h>
+
+#undef DEBUG
+#define SET_VALUES_OPTION 1
+#define EXIT_OPTION 2
+
+using namespace hyro;
+using namespace signal_generator;
+using namespace std::string_literals;
+
+int main(void)
+{
+  hyro::LogConfig config;
+  config.level = hyro::LogLevel::CRITICAL;
```

Comment:
So you already set the log level. Why do you also use DEBUG precompiler definition?